### PR TITLE
Feature/add login functionality

### PR DESCRIPTION
**Pull request by lilianaweimer**
Some of this happened before this PR because of a git mistake I made, which has been resolved. However, I thought I should include it anyway because my PR buddy hasn't seen any of my code yet.

## Description
Went a bit overboard, and did a lot more than just add login functionality:
- Added HTML
- Styled the page
- Added login functionality
- Created the traveler dashboard
- Created the travel agent dashboard
- Began working on displaying POST-ed trips to the traveler's trips
- I created `scripts.js` and `scripts-test.js` early on when I was confused and am going to delete them, they aren't necessary

## Motivation and Context

This completes iteration one and two, and starts on iteration three. I am still missing my spies, but am going to add those tomorrow. I also am unsure if the POST and DELETE should be methods on the classes or if I should keep them where they are. 

## How Has This Been Tested?

I have built the tests for the classes, but I haven't done the spies yet. The last time I checked all the tests were passing. I also checked the API and the trips have POST-ed successfully.

## Screenshots (if appropriate):
N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

I am planning to work on the README on Monday and Tuesday. 

The questions I have for my PR buddy are:
- Do you think that I should move POSTs and DELETEs to the classes? I'm not clear on this from the spec. 
- Do the classes seem like they are separate enough from each other? Do you have any advice if they don't?
- I have more global variables than I think I should, do you have any advice for moving some of them inside of things?

### CODE REVIEW

File: test/travelagent-test.js [hunk 1]
```diff
@@ -113,7 +114,7 @@ describe('Travel Agent', function() {
 
     beforeEach(() => {
     user = new User(userData);
```
Comment (jeremiahblackol) [inline test/travelagent-test.js]: does user hold all data?

File: src/index.js [hunk 2]
```diff
@@ -2,9 +2,89 @@
...
+        return travelAgent;
+    } else {
+        let userInfo = finduserByID(newID);
+        user = new User(userInfo);
+        traveler = new Traveler(newID, userInfo);
+        return traveler;
+    }
+}
+
+const checkLogInStatus = () => {
```
Comment (jeremiahblackol) [inline src/index.js:101]: what exactly is the logged in status for? if you instantiate a particular user based in id, does it need it?

File: test/travelagent-test.js (hunk 1 above)
Comment (lilianaweimer) [reply test/travelagent-test.js]: user just holds the password and whether they are logged in. I might change it later, it does seem kind of redundant.

File: src/index.js (hunk 2 above)
Comment (lilianaweimer) [reply src/index.js:101]: The check log in status determines whether the user has entered the correct password and shows the dashboard if they have. I'm not sure if it needs it, I might rethink some of my class structure later to simplify it.

**Pull request merged**